src/txadmission.cpp

```diff
@@ -654,12 +654,8 @@ bool ParallelAcceptToMemoryPool(Snapshot &ss,
         (IsNov2018Activated(chainparams.GetConsensus(), chainActive.Tip())) ? SCRIPT_ENABLE_CHECKDATASIG : 0;
     const uint32_t schnorrflag =
         (IsMay2019Activated(chainparams.GetConsensus(), chainActive.Tip())) ? SCRIPT_ENABLE_SCHNORR : 0;
-    const uint32_t segwit_flag =
-        (IsMay2019Activated(chainparams.GetConsensus(), chainActive.Tip()) && !fRequireStandard) ?
-            SCRIPT_ALLOW_SEGWIT_RECOVERY :
-            0;
 
-    const uint32_t featureFlags = cds_flag | schnorrflag | segwit_flag;
+    const uint32_t featureFlags = cds_flag | schnorrflag;
```

Comment:
It looks like the logic here changed from _allowing segwit recovery when non-standard transactions are accepted_ into mempool, to _never allow segwit recovery transactions to mempool_.

This is also tested in the QA test.

Just pointing this out in case this was not intentional.

Reply:
AFAIU the aim of the D3194 is to accept historical blocks that contains SegWit recovery txns, whereas at the same time such txns won't be accepted in the mempool.

Reply:
As was the old behavior, but additionally it now also rejects segwit recovery transactions when `acceptnonstdtxn=1` is set.

I think this new behavior is fine. Allowing non-standard transactions to be mined should not imply allowing segwitted coins to be stolen.

Reply:
Yes, agreed. with this changes there's no way to get segwit recovery txs transaction accepted in your mempool.

Reply:
I disagree with this change.  If the feature was added in consensus we should support it.  What is probably going on is that certain miners are being provided a custom release in the hopes that only they can recover segwit transactions.  We should not make a change that would deny our miners revenue unless they jump thru hoops.

Reply:
@gandrewstone 
so you are proposing to accept segwit recovery txns in the mempool in case `acceptnonstdtxn=1`?

I've no problem since either way we are going to accept blocks accepting blocks that contains this kind of transaction, so there's no risk of chain split in case we diverge from ABC in that particular regard.

What do you think @dagurval?

Reply:
It's more correct and clean to keep old behavior, but I don't like it.

Reply:
Ok I'm going to remove this code change so that we could still accept segwit recovery transaction by setting `acceptnonstdtxn=1`

Reply:
done